spring-cloud-schema-registry-core/src/main/java/org/springframework/cloud/schema/registry/controllers/ServerController.java

```diff
@@ -276,6 +300,39 @@ public String schemaDeletionNotPermitted(SchemaDeletionNotAllowedException ex) {
 		return errorMessage("Schema deletion is not permitted", ex);
 	}
 
+	private List<Schema> schemaReferenceResolver(String schemaReferenceHeader, String format) {
+		List<Schema> schemaReferences = new ArrayList<>();
+		Schema schemaReference;
+		String subject;
+		Integer version;
+		Matcher schemaReferenceMatcher;
+		for (String schemaReferenceEntry : schemaReferenceHeader.split(";")) {
```

Comment:
Method needs some refactoring.

Can you advice any "springy" way of doing schema reference resolution?